### PR TITLE
Fix core crash - Closes #7368, #7447 and #7591

### LINKED ISSUE #7368: Lisk Core crashes
### Expected behavior
Lisk Core shouldn't crash when Lisk Service runs against it.

### Actual behavior
Lisk Core crashes with the following error as soon as Lisk Service is started against it.

```
/root/lisk-core/node_modules/lisk-framework/dist-node/abi_handler/abi_client.js:20
        reject(new Error(message !== null && message !== void 0 ? message : `Timed out in ${ms}ms.`));
               ^

Error: Response not received in 500ms
    at Timeout.<anonymous> (/root/lisk-core/node_modules/lisk-framework/dist-node/abi_handler/abi_client.js:20:16)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
```

Other errors logged with subsequent node restarts are:
```
Error: unknown data not found error `node_hash does not exist`
    at ABIClient._listenToRPCResponse (/root/lisk-core/node_modules/lisk-framework/dist-node/abi_handler/abi_client.js:141:33)
/root/lisk-core/node_modules/lisk-framework/dist-node/abi_handler/abi_client.js:141
                deferred.reject(new Error(response.error.message));
                                ^
```

```
Error: unknown data not found error `node_hash does not exist`
    at ABIClient._listenToRPCResponse (/root/lisk-core/node_modules/lisk-framework/dist-node/abi_handler/abi_client.js:141:33)
/root/lisk-core/node_modules/@liskhq/lisk-codec/dist-node/keys.js:12
    throw new Error('Value yields unsupported wireType');
          ^

Error: Value yields unsupported wireType
    at Object.readKey (/root/lisk-core/node_modules/@liskhq/lisk-codec/dist-node/keys.js:12:11)
    at Object.readObject (/root/lisk-core/node_modules/@liskhq/lisk-codec/dist-node/collection.js:109:38)
    at Object.readArray (/root/lisk-core/node_modules/@liskhq/lisk-codec/dist-node/collection.js:149:51)
    at Object.readObject (/root/lisk-core/node_modules/@liskhq/lisk-codec/dist-node/collection.js:80:51)
    at Codec.decode (/root/lisk-core/node_modules/@liskhq/lisk-codec/dist-node/codec.js:49:36)
    at ABIClient._call (/root/lisk-core/node_modules/lisk-framework/dist-node/abi_handler/abi_client.js:120:96)
    at async Generator._generateBlock (/root/lisk-core/node_modules/lisk-framework/dist-node/engine/generator/generator.js:306:28)
    at async Generator._generateLoop (/root/lisk-core/node_modules/lisk-framework/dist-node/engine/generator/generator.js:264:32)
    at async Timeout._onTimeout (/root/lisk-core/node_modules/@liskhq/lisk-utils/dist-node/job_handlers/scheduler.js:26:17)
/root/lisk-core/node_modules/lisk-framework/dist-node/abi_handler/abi_client.js:141
 ```

### Steps to reproduce
- Clone Lisk Core branch: `release/4.0.0`
- `nvm use`
- `npm install --registry https://npm.lisk.com`
- `nvm run build`
- `./bin/run start -n devnet --api-ipc --api-ws --console-log info`
- Clone Lisk Service branch: `feature/blockchain-agnostic`
- Replace Line 9 in `jenkins/redis/docker-compose.yml` with `      - "6379:6379"`
- `cd jenkins/mysql && make up && cd ../redis && make up && cd ../..`
- `pm2 start ecosystem.config.js`

### Which version(s) does this affect? (Environment, OS, etc...)

SDK v6.0.0-alpha.2

### LINKED ISSUE #7447: Fail to execute block error
### Expected behavior
Lisk Core should run without any errors.

### Actual behavior

```
10:17:47 INFO lisk-framework: Generated new block (module=engine)
{
 "id": "c35965c42a708158cfe62e18a509e404e8cb24a3d5afbe0113784b11c8b41105",
 "height": 17,
 "generatorAddress": "cd56330913e4517f35cf689e849f5c208ed48b8e"
} 
10:17:47 ERROR lisk-framework: Fail to respond (module=lisk:app)
Message: Invalid context id 7b226572726f72223a7b226d657373616765223a2254686520616374696f6e20276765744e6f6465496e666f27206f6e206d6f64756c6520276170702720646f6573206e6f742065786973742e227d7d. Context is not initialized or different. 
 Trace: Error: Invalid context id 7b226572726f72223a7b226d657373616765223a2254686520616374696f6e20276765744e6f6465496e666f27206f6e206d6f64756c6520276170702720646f6573206e6f742065786973742e227d7d. Context is not initialized or different.
    at ABIHandler.verifyAssets (/Users/sameer/Documents/Lisk/github/lisk-core/node_modules/lisk-framework/src/abi_handler/abi_handler.ts:307:10)
    at ABIServer._listenToRequest (/Users/sameer/Documents/Lisk/github/lisk-core/node_modules/lisk-framework/src/abi_handler/abi_server.ts:198:32) 
{
 "method": "verifyAssets"
} 
10:17:47 ERROR lisk-framework: Fail to execute block. (module=engine)
Message: Response not received in 500ms 
 Trace: Error: Response not received in 500ms
    at Timeout.<anonymous> (/Users/sameer/Documents/Lisk/github/lisk-core/node_modules/lisk-framework/dist-node/abi_handler/abi_client.js:20:16)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7) 
10:17:48 INFO lisk-framework: Generated new block (module=engine)
{
 "id": "9936d05ac0b2859da8f90e523cddf5986c1eb6a4714790b3b79cf0de8495ec18",
 "height": 17,
 "generatorAddress": "cd56330913e4517f35cf689e849f5c208ed48b8e"
}
```

### Steps to reproduce

- Checkout Lisk Core tag: `v4.0.0-alpha.1`
- Comment out L137-146 in `src/application/modules/legacy/module.ts`
- `npm install --registry https://npm.lisk.com`
- `npm run build`
- Run Lisk Core: `rm -rf ~/.lisk && ./bin/run start --network devnet --api-ipc --api-ws --console-log info`
- Run Lisk Service from the branch `feature/blockchain-agnostic` against it

### Which version(s) does this affect? (Environment, OS, etc...)
Lisk SDK `v6.0.0-alpha.2`

### LINKED ISSUE #7591: Functional test is unstable
### Expected behavior
Functional test should be reliably passing to make PR smoothly reviewed and merged

### Actual behavior
Functional test is unstable and easy to be failing

### Steps to reproduce
Run jenkins multiple time

### Which version(s) does this affect? (Environment, OS, etc...)
alpha.3-

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #7368 , resolves #7447 and resolves #7591

1. In case of application success to commit but engine fails, it leaves the data at inconsistent state
2. If genesis state execution takes too long, it leaves the data at inconsistent state
3. ABI request was not counting the ID, so if there are multiple calls at the same time, req/res will be mixed
4. `oclif-dev` command was outdated

### How was it solved?

- Update ABI call from `ready` to `init` and add state revert logic in case of inconsistent state
- Update `oclif-dev` to use `oclif` command
- Update ABI client to count the ID

### How was it tested?

- Run lisk-core and lisk-service together
or
- Run dpos-mainchain and make multiple RPC request and see it doesnt crash

### CODE REVIEW

File: commander/package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
 		"test:coverage": "jest --coverage=true --coverage-reporters=text",
 		"test:ci": "jest --coverage=true --coverage-reporters=json --verbose",
 		"postbuild": "copyfiles -u 1 -a 'src/bootstrapping/templates/lisk-template-ts/templates/**/*' dist/",
-		"prepack": "oclif-dev manifest"
+		"prepack": "oclif manifest"
```
[the remark this answers is not in the document]
Comment (shuse2) [reply commander/package.json:46]: this only executed in publication, so testing this fails without this =)

File: framework/src/abi_handler/abi_handler.ts [hunk 2]
```diff
@@ -118,9 +119,90 @@ export class ABIHandler implements ABI {
 		return this._chainID;
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
-	public async ready(_req: ReadyRequest): Promise<ReadyResponse> {
+	public async cacheGenesisState(): Promise<void> {
+		const { version, root } = await this._stateDB.getCurrentState();
+		// Slip if state db is already initialized
```
Comment (ishantiw) [inline framework/src/abi_handler/abi_handler.ts:124]:
```suggestion
		// Skip if state db is already initialized
```

File: framework/src/abi_handler/abi_handler.ts [hunk 3]
```diff
@@ -118,9 +119,90 @@ export class ABIHandler implements ABI {
...
+		if (version !== 0 || !root.equals(utils.hash(Buffer.alloc(0)))) {
+			this._logger.debug(
+				{ version, root: root.toString('hex') },
+				'Skip caching genesis state. state is already initialized',
```
Comment (ishantiw) [inline framework/src/abi_handler/abi_handler.ts:128]:
```suggestion
				'Skip caching genesis state as it is already initialized',
```

File: framework/src/abi_handler/abi_handler.ts [hunk 4]
```diff
@@ -118,9 +119,90 @@ export class ABIHandler implements ABI {
...
+			'Cached genesis state',
+		);
+	}
+
+	public async init(req: InitRequest): Promise<InitResponse> {
 		this._chainID = Buffer.from(this._config.genesis.chainID, 'hex');
+		const currentState = await this._stateDB.getCurrentState();
+		if (req.lastBlockHeight > currentState.version) {
+			throw new Error(
+				`Invalid engine state. Conflict at height Engine: ${req.lastBlockHeight} and application state ${currentState.version}.`,
```
Comment (ishantiw) [inline framework/src/abi_handler/abi_handler.ts:182]: Shouldn't this be?
```suggestion
				`Invalid engine state. Conflict at engine height: ${req.lastBlockHeight} and application state ${currentState.version}.`,
```

File: framework/src/abi_handler/abi_handler.ts [hunk 5]
```diff
@@ -363,14 +452,23 @@ export class ABIHandler implements ABI {
 				)}. Context is not initialized or different.`,
 			);
 		}
+		// eslint-disable-next-line
```
Comment (ishantiw) [inline framework/src/abi_handler/abi_handler.ts:455]: Can we be specific what is disabled on the next line?

File: framework/src/abi_handler/abi_client.ts [hunk 6]
```diff
@@ -294,21 +295,31 @@ export class ABIClient implements ABI {
...
+			this._globalID = BigInt(0);
+		}
 
 		const resp = await Promise.race([
 			response.promise,
 			timeout(DEFAULT_TIMEOUT, `Response not received in ${DEFAULT_TIMEOUT}ms`),
 		]);
+		this._logger.debug(
+			{ method: requestBody.method, id: requestBody.id, file: 'abi_client' },
+			'Reseived response from ABI server',
```
Comment (ishantiw) [inline framework/src/abi_handler/abi_client.ts:318]:
```suggestion
			'Received response from ABI server',
```

**Review by ishantiw — CHANGES_REQUESTED** [review]

**Review by ishantiw — APPROVED** [review]
🎉

File: framework/test/unit/abi_handler/abi_handler.spec.ts [hunk 7]
```diff
@@ -79,12 +79,56 @@ describe('abi handler', () => {
...
+		it('should reject init if states are different with the same height', async () => {
+			(stateDBMock.getCurrentState as jest.Mock).mockResolvedValue({ root, version: 21 });
+
+			(stateDBMock.revert as jest.Mock).mockResolvedValue(root);
+
+			await expect(
+				abiHandler.init({
+					chainID: Buffer.alloc(4),
+					lastBlockHeight: 19,
+					lastStateRoot: utils.getRandomBytes(32),
```
[the remark this answers is not in the document]
Comment (shuse2) [reply framework/test/unit/abi_handler/abi_handler.spec.ts:110]: yes that is on purpose to test if it doesn't match, same for below

**Pull request merged**